pkg/graph/schema/types/model_to_operations_matrix.graphql

```diff
@@ -0,0 +1,95 @@
+# ModelsToOperationMatrix Notes:
+# - There's purposefully no `Milestones` query right now, since there's currently no need to query milestones oustide the scope of a Category
+type ModelsToOperationMatrix {
+  categories: [Category!]! # Categories is at the top level because you _can_ have empty categories
+  commonMilestones: [CommonMilestone!]!
+  solutions: [Solution!]! # Need to support solutions that _don't_ have milestones
+}
+
+# Milestone represents a Milestone that has been added to a Model Plan's MTO
+type Milestone {
+  # DB Fields
+  id: UUID!
+  name: String!
+  description: String! # Not sure if this is actually needed - I think there might only be descriptions on Milestones in the library? (CommonMilestone)
+  facilitatedBy: String # TODO: Make enum for `MilestoneFacilitator` (TBD if that's the same as `SolutionFacilitator`)
+  needBy: Time
+  status: String! # TODO: Make enum for `MilestoneStatus`
+  riskIndicator: String! # TODO: Make enum for `MTORiskIndicator`
+  isDraftMilestone: Boolean!
+
+  # Custom Resolvers
+  addedFromMilestoneLibrary: Boolean!
+  solutions: [Solution!]!
+  category: Category # Can be NULL if not part of a category (TODO: Unless we handle "Uncategorized" as an actual category)
+  # TODO Is ^^ confusing for fetching subcategories too? Should we just have 2 resolvers, one for `category`, one for `subcategory`?
+
+}
+
+# Solution represents a Solution that has been added to a Model Plan's MTO
+type Solution {
+  # DB Fields
+  id: UUID!
+  name: String!
+  description: String!
+  facilitatedBy: String # TODO: Make enum for `SolutionFacilitator` (TBD if that's the same as `MilestoneFacilitator`)
+  status: String! # TODO: Make enum for `SolutionStatus`
+  riskIndicator: String! # TODO: Make enum for `MTORiskIndicator`
+
+  # Custom Resolvers
+  relatedMilestones: [Milestone!]! # Can be empty, since you can have a solution with no milestones
+  addedFromMilestoneLibrary: Boolean! # TRUE if this is linked to a common milestone
+  solutions: [Solution!]!
+}
+
+# CommonMilestone represents a Milestone from the "Common Milestones" library
+# It should not be used in contexts where a milestone has been added to a Model Plan (at that point, it'll be a "Milestone", not a "CommonMilestone")
+type CommonMilestone {
+  # DB Fields
+  id: UUID! # TODO Is this really needed? It might be kind of confusing to reference `id` like this
+  key: CommonMilestoneKey! # Similar to OperationalSolutionKey, enum of possible common milestones
+  name: String!
+  description: String!
+
+  # Custom Resolvers
+  isAdded: Boolean! # To represent if this Common Milestone is _already_ part of the Model Plan's MTO
```

Comment:
I'm interested in this field.  At it's simplest, we just need to have knowledge of the model plan to know if they milestone was already added or not correct?

From lessons learned from operational solutions, I wonder if we should think about Unions to return common milestones?

 Would it be useful to have a resolver to get the milestone here by model_plan_id and common_milestone key?

Reply:
I think this is probably covered from our call, but for posterity:

Since we only query (currently) common milestones _in the context_ of a Model Plan, we can resolve this `isAdded` based on whether or not we see a Milestone with the same UUID/Fkey/whatever in that model plan's MTO!

Reply:
But feel free to un-resolve if you've got any more questions/thoughts on this 😁